packages/@uppy/companion/src/server/b2.js

```diff
@@ -0,0 +1,170 @@
+const request = require('request')
+
+const B2_API_VERSION = 2
+const B2_API_URL = `https://api.backblazeb2.com/b2api/v${B2_API_VERSION}/`
+
+module.exports = class B2Client {
+  constructor (options) {
+    this.applicationKeyId = options.credentials.applicationKeyId
+    this.applicationKey = options.credentials.applicationKey
+
+    this.retries = 3 // TODO
+
+    // This will be populated after a successful authorization
+    this.auth = {}
+  }
+
+  getURL (method) {
+    if (!this.auth.apiUrl) {
+      throw new Error('getURL() called before successful authorize()')
+    }
+    return this.auth.apiUrl + `/b2api/v${B2_API_VERSION}/${method}`
+  }
+
+  /**
+   * Request new authorization token from Backblaze.
+   */
+  authorize () {
+    return this._authorizing || (this._authorizing = new Promise((resolve, reject) => {
```

Comment:
I think this pattern is a bit confusing—could we use one of these instead?
```js
if (this._authorizing) return this._authorizing
this._authorizing = new Promise(...)
return this._authorizing
```
or
```js
if (!this._authorizing) {
  this._authorizing = new Promise(...)
}
return this._authorizing
```